### PR TITLE
HWP-404: Frontend - Community Enhancements

### PR DESCRIPTION
**Pull request by BradyMitch**
# Description

This PR includes the following proposed change(s):

- Updated styles of communities page.
- Added latestActivity to communities in profile page.
- Added member count to communities.
- Changes to how join/leave community buttons show.
- Changed singular post to show local time.
- Changed ./mongo/init communities.json for Welcome community to add createdOn field.

## Requires

Add 'Requires Attention' label if appropriate.

- [ ] Requires an update to the .env file
- [ ] Requires adding a file listed in the .gitignore
- [ ] Requires an opinion or answer to a question, see comments.
- [x] Requires other attention, see below.

If there are any requirements for the developer to make after this PR is merged please list them here.

If you didn't see in the other PR for HWP-404. You need to remove the files inside `./mongo/data` directory. If you don't there will be some unexpected behavior. 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Tested alongside HWP-404 backend changes.

For pull requests that are not just a simple fix, please check that the branch works on your local machine.

- [x] Tested by Zach.
- [ ] Tested by Brandon.
- [x] Tested by Brady.

## Does the change impact or break the Docker build?

- [ ] Yes
- [x] No

If Yes: Has Docker been updated accordingly?

- [ ] Yes
- [ ] No

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] New and existing unit tests pass locally with my changes
- [ ] If attention is required by the developer such as updating the .env file, these requirements have been listed.

### CODE REVIEW

**Comment by BradyMitch** [comment]
@akroon3r I created a ticket HWP-413 for you. Please drop the database data.

File: app/client/src/components/Community.jsx [hunk 1]
```diff
@@ -11,16 +13,19 @@ import {
   MenuList,
   Typography,
 } from "@mui/material";
-import FlagTwoToneIcon from "@mui/icons-material/FlagTwoTone";
+// import FlagTwoToneIcon from "@mui/icons-material/FlagTwoTone";
```
Comment (akroon3r) [inline app/client/src/components/Community.jsx:16]: if we don't need this, please remove

File: app/client/src/components/Community.jsx [hunk 2]
```diff
@@ -29,10 +34,10 @@ const Community = (props) => {
 
   const { community } = props;
 
-  const handleFlagCommunityClick = () => {
-    props.openFlagCommunityModal(community);
-    handleMenuClose();
-  };
+  // const handleFlagCommunityClick = () => {
```
Comment (akroon3r) [inline app/client/src/components/Community.jsx:37]: if we don't need this, please remove

File: app/client/src/components/SingularPost.jsx [hunk 3]
```diff
@@ -39,7 +39,7 @@ const SingularPost = (props) => {
 
   useEffect(() => {
     props.post._id === id || props.getPost(id);
-  }, []);
+  });
```
Comment (ZachBourque) [inline app/client/src/components/SingularPost.jsx:42]: This change needs to be undone

File: mongo/init/data/communities.json [hunk 4]
```diff
@@ -3,13 +3,7 @@
...
     "description": "Say hi!",
     "creator": "",
-    "members": [],
-    "__v": 0
-  },
-  {
-    "title": "Test",
-    "description": "Many wow",
-    "creator": "",
+    "createdOn": "June 15th 2022, 9:21:42 pm",
```
[the remark this answers is not in the document]
Comment (BradyMitch) [reply mongo/init/data/communities.json:6]: It's for the welcome community only. I think I could add the initialization of data to the backend, completely eliminating the need for the `./mongo/init` directory and allowing me to dynamically assign a date. I'll create a ticket for this and add it to backlog.

**Comment by ZachBourque** [comment]
Tested. Works well, one small optimization would be to modify the number of members in the redux store when joining a community. This change would immediately display to the change to the user.

Other than that, works great!

**Review by ZachBourque — APPROVED** [review]

File: app/client/src/components/Community.jsx (hunk 1 above)
Comment (ZachBourque) [reply app/client/src/components/Community.jsx:16]: Exactly. I added stuff in to be used later

**Pull request merged**